### ISSUE TITLE
extra files are passed to AVA when following after another fly task

### ISSUE THREAD

**Issue by frantic1048** [pine613/fly-ava#13]
Let's say we have such a `flyfile.js`

```js
exports.foo = function* foo (fly) {
  yield fly.source('./src/index.js').target('build/index.js')
}

exports.test = function* test (fly) {
  yield fly.start('foo')
  yield fly.source('./test/index.js').ava() // error
}

exports.test_another = function* test_another (fly) {
  yield fly.source('./src/index.js').target('build/index.js')
  yield fly.source('./test/index.js').ava() // works
}
```

`./src/index.js` is just a blank file.

`./test/index.js` is a regular AVA test file:

```js
import test from 'ava'

test('foo', t => {
  t.pass()
})
```

Fly will fail `test` task with error: `test failed because No tests found in src/index.js, make sure to import "ava" at the top of your test file`. Even I just source('./test/index.js') to AVA.

But `task_another` runs well.

I'm not sure if this is an fly issue.

**Comment by lukeed** [comment]
It's a Fly bug that's been on my todo list. 

When you run tasks from inside other tasks, there's a rare chance that the source files will cross over. I think I know why, but haven't had time to do it yet. 

For now, either define your task order via the CLI/NPM command definition:

```js
{
  "test": "fly build test"
}
```

Or use `fly.serial` or `fly.parallel` instead of `fly.start` inside your task.

**Comment by frantic1048** [comment]
Thanks for the explaination, I know how to workaround this.

By the way, I found `fly.serial` also make the source files cross over, like this:

```js
exports.test = function* test (fly) {
  yield fly.serial(['foo'])
  yield fly.source('./test/index.js').ava() // error
}
```

**Comment by lukeed** [comment]
Yeah, sorry. The **best** and most reliable way is to chain your commands from the NPM script.

```js
{
  "test": "fly foo test"
}
```

**Comment by pine** [comment]
Is this the `fly-ava` plugin bug? @frantic1048 @lukeed

**Comment by lukeed** [comment]
@pine Nope, this is a Fly bug.

**Comment by pine** [comment]
OK! Thanks.
